python/dicom_archive.py

```diff
@@ -0,0 +1,325 @@
+
+parser.add_argument(
+    '--year',
+    action='store_true',
+    help='Create the archive in a year subdirectory (example: 2024/DCM_2024-08-27_FooBar.tar)')
+
+parser.add_argument(
+    '--overwrite',
+    action='store_true',
+    help='Overwrite the DICOM archive file if it already exists')
+
+parser.add_argument(
+    '--db-insert',
+    action='store_true',
+    help=(
+        'Insert the created DICOM archive in the database (requires the archive '
+        'to not be already inserted)'))
+
+parser.add_argument(
+    '--db-update',
+    action='store_true',
+    help=(
+        'Update the DICOM archive in the database, which requires the archive to be '
+        'already be inserted), generally used with \'--overwrite\''))
+
+parser.add_argument(
+    'source',
+    help='The source DICOM directory')
+
+parser.add_argument(
+    'target',
+    help='The target directory for the DICOM archive')
+
+args = parser.parse_args()
+
+# Typed arguments
+
+profile:   str | None = args.profile
+source:    str        = args.source
+target:    str        = args.target
+verbose:   bool       = args.verbose
+today:     bool       = args.today
+year:      bool       = args.year
+overwrite: bool       = args.overwrite
+db_insert: bool       = args.db_insert
+db_update: bool       = args.db_update
+
+# Check arguments
+
+if db_insert and db_update:
+    print_error_exit(
+        'Arguments \'--db-insert\' and \'--db-update\' must not be set both at the same time.',
+        lib.exitcode.INVALID_ARG,
+    )
+
+if (db_insert or db_update) and not profile:
+    print_error_exit(
+        'Argument \'--profile\' must be set when a \'--db-*\' argument is set.',
+        lib.exitcode.INVALID_ARG,
+    )
+
+if not os.path.isdir(source) or not os.access(source, os.R_OK):
+    print_error_exit(
+        'Argument \'--source\' must be a readable directory path.',
+        lib.exitcode.INVALID_ARG,
+    )
+
+if not os.path.isdir(target) or not os.access(target, os.W_OK):
+    print_error_exit(
+        'Argument \'--target\' must be a writable directory path.',
+        lib.exitcode.INVALID_ARG,
+    )
+
+# Connect to database (if needed)
+
+db = None
+if profile is not None:
+    db = Database(load_config_file(profile).mysql, False)
+    db.connect()
+
+# Check paths
+
+while source.endswith('/'):
+    source = source[:-1]
+
+base_name = os.path.basename(source)
+
+tar_path     = f'{target}/{base_name}.tar'
+zip_path     = f'{target}/{base_name}.tar.gz'
+summary_path = f'{target}/{base_name}.meta'
+log_path     = f'{target}/{base_name}.log'
+
+check_create_file(tar_path)
+check_create_file(zip_path)
+check_create_file(summary_path)
+check_create_file(log_path)
+
+print('Extracting DICOM information (may take a long time)')
+
+summary = lib.dicom.summary_make.make(source, verbose)
+
+if db is not None:
+    print('Checking database presence')
+
+    archive = lib.dicom.dicom_database.get_archive_with_study_uid(db, summary.info.study_uid)
```

Comment:
Maybe rename to db_archive to make is clear that this is the archive found in the DB?

```suggestion
    db_archive = lib.dicom.dicom_database.get_archive_with_study_uid(db, summary.info.study_uid)
```

Reply:
That makes sense, done !

Reply:
Well, named `dicom_archive` but it still kind of works when compared with the other variable names IMO.